### PR TITLE
WELD-2196 Replace Guava - caches

### PR DESCRIPTION
**Pull request by mkouba**

### CODE REVIEW

File: impl/src/main/java/org/jboss/weld/util/cache/ComputingCache.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+ */
+package org.jboss.weld.util.cache;
+
+import org.jboss.weld.util.Consumer;
+
+/**
+ * A simple abstraction for computing cache.
+ * <p>
+ * Implementations are not required to support recursive computations. A cache client must be aware that such computations may result in livelocks, inifinite
+ * loops and other undesired situations.
```
Comment (tremes) [inline impl/src/main/java/org/jboss/weld/util/cache/ComputingCache.java]: Typo in the word "infinite."

File: impl/src/main/java/org/jboss/weld/util/cache/ComputingCache.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+     *
+     * @return an immutable iterable of values
+     */
+    Iterable<V> getAllPresentValues();
+
+    /**
+     * Performs the given action for each cached value.
+     * @param consumer the given action
+     */
+    void forEachValue(Consumer<? super V> consumer);
```
Comment (tremes) [inline impl/src/main/java/org/jboss/weld/util/cache/ComputingCache.java]: This method seems to be never used so far..

Comment (mkouba) [inline impl/src/main/java/org/jboss/weld/util/cache/ComputingCache.java]: You're right, I will remove this method.

**Comment by mkouba** [comment]
Merged.

**Pull request closed without merging**